### PR TITLE
Custom quotedIdentifier and concatNullYields connection property

### PR DESCRIPTION
**Pull request by lilgreenbird**
for review only, not to be merged

### CODE REVIEW

File: src/test/java/com/microsoft/sqlserver/jdbc/SQLServerConnectionTest.java [hunk 1]
```diff
@@ -319,6 +319,12 @@ public void testDataSource() throws SQLServerException {
 
         ds.setKeyStorePrincipalId(stringPropValue);
         assertTrue(ds.getKeyStorePrincipalId().equals(stringPropValue));
+        
+        ds.setQuotedIdentifier(stringPropValue);
+        assertTrue(ds.getQuotedIdentifier().equals(stringPropValue));
+        
+        ds.setConcatNullYieldsNull(stringPropValue);
+        assertTrue(ds.getConcatNullYieldsNull().equals(stringPropValue));
     }
```
Comment (machavan) [inline src/test/java/com/microsoft/sqlserver/jdbc/SQLServerConnectionTest.java:328]: Would it be good to also have a test that will test the after-effect of setting these options by performing few database options and validating the data. E.g. doing an insert select that concatenates nulls to fields etc.

**Comment by David-Engel** [comment]
One goal is to see how our tests behave with the options set to their non-default values. I assume everything will pass. The only small risk area in my mind is the metadata queries so ensuring those run with the non-default values is more important to me. You can simply change the default values or the properties in the PR to make testing easier.

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java [hunk 2]
```diff
@@ -3269,6 +3269,35 @@ else if (0 == requestedPacketSize)
...
             }
 
+            // check QUOTED_IDENTIFIER property
+            String quotedIdentifierProperty = SQLServerDriverStringProperty.QUOTED_IDENTIFIER.toString();
+            String quotedIdentifierValue = activeConnectionProperties.getProperty(quotedIdentifierProperty);
+            if (null == quotedIdentifierValue) {
+                quotedIdentifierValue = SQLServerDriverStringProperty.QUOTED_IDENTIFIER.getDefaultValue();
+                activeConnectionProperties.setProperty(quotedIdentifierProperty, quotedIdentifierValue);
+            }
+            String quotedIdentifierOption = OnOffOption.valueOfString(quotedIdentifierValue).toString();
```
[the remark this answers is not in the document]
Comment (lilgreenbird) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java:3279]: this is already taken care of when we parse the Url the properties all come in here in the form "prop=value"

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java [hunk 3]
```diff
@@ -720,6 +722,47 @@ public String toString() {
...
+        this.option = option;
+    }
+
+    @Override
+    public String toString() {
+        return option;
+    }
+
+    static OnOffOption valueOfString(String value) throws SQLServerException {
+        OnOffOption option = null;
```
[the remark this answers is not in the document]
Comment (lilgreenbird) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java:741]: see above

**Comment by tkyc** [comment]
When a reconnect thread (for ICR) executes a TDSCommand, it needs to create a TDSCommand counter otherwise it will error out. We should add a fix for counter initialization for TDSCommands in the reconnect thread in the main branch.

**Pull request closed without merging**